cmd/ingress-operator/render.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+)
+
+func render(dir string, prefix string) error {
+	files := []string{
+		manifests.CustomResourceDefinitionManifest,
+		manifests.NamespaceManifest,
+	}
+
+	if dir == "-" {
+		for _, file := range files {
+			fmt.Printf("# file: %s\n", filepath.Base(file))
+			fmt.Println(manifests.MustAssetString(file))
+			fmt.Println("---")
+		}
+		return nil
+	}
+
+	if err := os.MkdirAll(dir, 0666); err != nil {
+		return fmt.Errorf("failed to create output directory %q: %v", dir, err)
+	}
+
+	for _, file := range files {
+		outputFile := filepath.Join(dir, prefix+filepath.Base(file))
```

Comment:
```suggestion
		outputFile := filepath.Join(filepath.Join(dir, prefix), filepath.Base(file))
```

Reply:
I don't think that's the same. The prefix is supposed to be prepended to the file base name, not the output directory.

Reply:
Ah, right, sorry.  `filepath.Join` would be inappropriate because the prefix is part of the filename, not a separate path component.